src/analysis.h

```diff
@@ -736,58 +736,94 @@ namespace Faunus {
         template<class Tspace>
             class ChargeFluctuations : public Analysisbase {
                 const Tspace& spc;
+                typedef typename Tspace::Tpvec Tpvec;
                 std::vector< std::map<int,int> > idcnt; // populations of types of atomic indexes
                 std::vector< Average<double> > charge; // average charges of atomic indexes
-                std::vector<size_t> indexes; // lower and upper bound of indexes to analyze
-                size_t ndx0, ndx1, cnt;
+                std::string molname; // name of molecule to analyze
+                int molid; // id of molecule to analyze
+                bool verbose = false;
+                size_t cnt;
 
                 void _sample() override {
-                    cnt = 0;
-                    for (auto p=spc.p.begin()+ndx0; p!=(spc.p.begin()+ndx1+1); ++p) {
-                        idcnt.at(cnt)[p->id]++;
-                        charge.at(cnt) += p->charge;
-                        ++cnt;
+                    for (auto &g : spc.groups) {
+                        cnt = 0;
+                        if (g.id == molid) {
+                            for ( auto p = g.begin(); p < g.trueend() ; ++p) {
```

Comment:
Should this go to `trueend()`?

Reply:
No, I agree. I got confused and here it should be auto &p : g. Later in the code, when I save the PQR file I should instead loop to the trueend() to make sure that the whole molecule is saved.